src/executables/util_cs.c

```diff
@@ -3260,53 +3271,64 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
 	  strcat (local_database_name, "@localhost");
 
 	  db_clear_host_connected ();
-
-	  if (check_database_name (local_database_name))
+	  if ((check_applied_info != false) && check_database_name (local_database_name))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  if (db_login ("DBA", NULL) != NO_ERROR)
+	  if ((check_applied_info != false) && (db_login ("DBA", NULL) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  error = db_restart (arg->command_name, TRUE, local_database_name);
-	  if (error != NO_ERROR)
+	  if ((check_applied_info != false) && (db_restart (arg->command_name, TRUE, local_database_name) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
 
-	  if (HA_DISABLED ())
+	  if ((check_applied_info != false) && HA_DISABLED ())
 	    {
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      printf ("\nERROR : ");
 	      PRINT_AND_LOG_ERR_MSG (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_APPLYINFO,
 						     APPLYINFO_MSG_NOT_HA_MODE));
-	      goto check_applied_info_end;
 	    }
+	  error = NO_ERROR;
 
-	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
-				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
+	  error = la_log_page_check (local_database_name, log_path, pageid, &check_applied_info, &check_copied_info,
+				     &check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
-	  (void) db_shutdown ();
+	  if (error == NO_ERROR)
+	    {
+	      (void) db_shutdown ();
+	    }
 	}
       else if (check_copied_info)
 	{
 	  memset (local_database_name, 0x00, CUB_MAXHOSTNAMELEN);
 	  strcpy (local_database_name, database_name);
 	  strcat (local_database_name, "@localhost");
 
-	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
-				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
+	  error = la_log_page_check (local_database_name, log_path, pageid, &check_applied_info, &check_copied_info,
+				     &check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
 	}
 
-    check_applied_info_end:
       if (error != NO_ERROR)
 	{
-	  PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	  PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
+	  check_applied_info = false;
```

Comment:
You want to prevent printing delay info when there is an error ?

Reply:
Right. the case of invalid -L argument .

Reply:
It's intended. OK.